samcli/local/lambdafn/runtime.py

```diff
@@ -108,6 +108,12 @@ def invoke(self,
                     timer.cancel()
                 self._container_manager.stop(container)
 
+                # Flush stdout and stderr streams so that logs from the invocation are written out
+                if stdout:
```

Comment:
A better place for this would be when we write the logs to stdout and stderr. This can be found here: https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/docker/container.py#L203

Placing the flushing here makes the code a little hard to maintain and reason about, since the `.flush` happens in a different place then the `.write`.